src/util/redisConnector.js

```diff
@@ -0,0 +1,36 @@
+const redis = require('redis');
+const log = require('../logger');
+
+let a = 1;
+const host = process.env.REDIS_HOST || 'localhost';
+const port = process.env.REDIS_PORT || 6379;
+const password = process.env.REDIS_PASSWORD || '';
+const DBConnector = {
+  redisInstance: redis.createClient({
+    socket: {
+      host,
+      port,
+    },
+    password,
+  }),
+  async getRedisInstance() {
+    if (!DBConnector.redisInstance?.isOpen) {
+      await DBConnector.redisInstance
+        .connect()
+        .then((res) => {
+          log.info("Redis Connected!");
+          a += 1;
+        })
+        .catch((err) => {
+          log.info(`err happened while connecting to redis ${err}`);
+        });
+    }
+    return DBConnector.redisInstance;
+  },
+};
+process.on('exit', async () => {
+  await DBConnector.redisInstance.quit().then((res) => {
+    log.info(`Redis Instance Terminated Gracefully ${res}`);
+  });
+});
```

Comment:
What do mean by kill signals. Like Terminating gracefully and other ?

Reply:
if you see for every kill signal gracefulShutdown is called which is making the process.exit() call https://github.com/rudderlabs/rudder-transformer/blob/0effcfa45e2fcdaa6df43578239f32ddbce9c4e8/src/index.js#L50 through force exit and through the same we are catching it here. 
However, the log at L37 is not printing so it might not be waiting for redis to quit. So that is one point I would like to discuss

Reply:
Ok lets discuss sometime today